webview/src/experiments/components/App.test.tsx

```diff
@@ -641,31 +639,57 @@ describe('App', () => {
 
       fireEvent.mouseLeave(testParamCell, { bubbles: true })
 
-      jest.advanceTimersByTime(1)
+      jest.advanceTimersByTime(CELL_TOOLTIP_DELAY)
       expect(screen.queryByRole('tooltip')).not.toBeInTheDocument()
-
-      jest.mocked(window.requestAnimationFrame).mockRestore()
     })
 
-    it('should show a tooltip with the full number on number cells', () => {
+    it('should show the expected tooltip for all data types', () => {
+      const expectTooltipValue: (args: {
+        cellLabel: string
+        expectedTooltipResult: string
+      }) => void = ({ cellLabel, expectedTooltipResult }) => {
+        const testParamCell = screen.getByText(cellLabel)
+        expect(testParamCell).toBeInTheDocument()
+
+        fireEvent.mouseEnter(testParamCell, { bubbles: true })
+
+        jest.advanceTimersByTime(CELL_TOOLTIP_DELAY)
+        const tooltip = screen.queryByRole('tooltip')
+        expect(tooltip).toHaveTextContent(expectedTooltipResult)
+
+        fireEvent.mouseLeave(testParamCell, { bubbles: true })
+
+        jest.advanceTimersByTime(CELL_TOOLTIP_DELAY)
+        expect(screen.queryByRole('tooltip')).not.toBeInTheDocument()
+      }
+
       render(<App />)
       fireEvent(
         window,
         new MessageEvent('message', {
           data: {
-            data: testData,
+            data: dataTypesTableData,
             type: MessageToWebviewType.SET_DATA
           }
         })
       )
 
-      const testMetricCell = screen.getByText('1.9293')
-      fireEvent.mouseEnter(testMetricCell, { bubbles: true })
-
-      jest.advanceTimersByTime(CELL_TOOLTIP_DELAY)
-      const tooltip = screen.getByRole('tooltip')
-      expect(tooltip).toBeInTheDocument()
-      expect(tooltip).toHaveTextContent(String(testMetricNumberValue))
+      expectTooltipValue({
+        cellLabel: '1.9293',
+        expectedTooltipResult: '1.9293040037155151'
+      })
+      expectTooltipValue({
+        cellLabel: 'true',
+        expectedTooltipResult: 'true'
+      })
+      expectTooltipValue({
+        cellLabel: 'false',
+        expectedTooltipResult: 'false'
+      })
+      expectTooltipValue({
+        cellLabel: '[true, false, string, 2]',
+        expectedTooltipResult: '[true, false, string, 2]'
+      })
```

Comment:
Multiple test titles would have more clean messages, but I don't have the heart to re-render the same fixture 3+ more times to do it.